### DISCUSSION TITLE
Does to_dense_batch retain order of edge_index?

### DISCUSSION THREAD

**Discussion by antoni4040** [pyg-team/pytorch_geometric#6277] [Q&A] [answered]
Weirdly specific question, when doing to_dense_batch like this:
`score, _ = to_dense_batch(score, batch[src], fill_value=float("-Inf"))`
Does it keep the order of the edges as they were before?
(part of my continuing efforts to build a GNN-PPO hybrid to select the best edge for a problem)
As in PPO I need to use the index of the previously taken action, I need to pass the same graph through the model at different times and retain the order of the edges as they were in the graph, even if it's part of a much larger batch.
Sorry for the question bombardment, I will contribute a tutorial if I ever figure it out. :)

**Comment by rusty1s** [comment, marked as answer]
Yes, the order is maintained. You can verify this based on some toy data.

**Reply by antoni4040** [reply]
Thank you!
